jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -2474,6 +2474,66 @@ ecma_object_class_is (ecma_object_t *object_p, /**< object */
   return false;
 } /* ecma_object_class_is */
 
+/**
+ * Checks if the given argument is an array or not.
+ *
+ * @return bool
+ */
+inline bool JERRY_ATTR_ALWAYS_INLINE
+ecma_object_is_array (ecma_value_t arg) /**< argument */
+{
+  if (ecma_is_value_object (arg))
```

Comment:
This can be written in the following way:

```c
return (ecma_is_value_object (arg)
        && ecma_get_object_type (ecma_get_object_from_value (arg)) == ECMA_OBJECT_TYPE_ARRAY);
```